src/libraries/System.IO.Compression/src/System/IO/Compression/ZipBlocks.cs

```diff
@@ -57,11 +56,11 @@ public static bool TryReadBlock(ReadOnlySpan<byte> bytes, out int bytesConsumed,
             }
 
             field._data = bytes.Slice(FieldLocations.DynamicData, field._size).ToArray();
-            bytesConsumed += field._size;
+            bytesConsumed = field.Size + SizeOfHeader;
```

Comment:
Good catch, we don't care about bytesConsumed if we return false. Better add the numbers only once.